torch/autograd/profiler.py

```diff
@@ -221,6 +223,14 @@ def __init__(
                 use_kineto
             ), "Device-only events supported only with Kineto (use_kineto=True)"
 
+        if self.use_device and self.use_device != _get_privateuse1_backend_name():
+            warn(f"{self.use_device} doesn't support profile.")
+            self.use_device = None
+
+        if self.use_device == "cuda":
```

Comment:
If use_device is cuda, won't it get set to none on line 228?

Reply:
Thanks for your opinion. if use_device is cuda, use_device indeed will be set to none! I will fix it now!